### DISCUSSION TITLE
Only One way traffic encrypted in Site-to-site ipsec tunnel configuration

### DISCUSSION THREAD

**Discussion by mkcdot** [strongswan/strongswan#2289] [Q&A]
I am setting up site-to-site tunnel between two LANs (LAN A: 192.168.0.0/24 and LAN B: 192.168.2.0/24) as per following example: [https://www.strongswan.org/testing/testresults/ikev2/net2net-cert/].
I am using pw-strongswan docker latest release. (however issue remains same with non-pqc strongswan with 
I have setup gateways for these LANs on two ubuntu 20.04 Computers, say Gateway-A and Gateway-B (with IPs 192.168.0.1 and 192.168.2.1 respectively used as gateway for above LANs). I have connected Gateway A and B in another LAN with IPs 192.168.47.198 and 192.168.47.199 respectively.
I am able to setup tunnel between these two LANs and SA status shows INSTALLED.

When I send ping command (or access webserver in LAN B) from any device in LAN A or from Gateway A to any device in LAN B or to Gateway B, there is no ESP packet seen in wireshark captures on both gateways. Only plain protocol packets are seen in wireshark.
But when I send ping command (or access webserver in LAN A) from any device in LAN B or from Gateway B to any device in LAN A or to Gateway A, ESP packets are seen in wireshark captures on both gateways.
However communication link is established between these two LANs in both directions i.e. any device in LAN A can access any device in LAN B and vice versa, when tunnel is established. And this communication link is lost when ipsec tunnel is terminated. (as expected)
Only issue remains is traffic is encrypted for LAN B devices but not LAN A devices as mentioned before.

What is wrong with my configuration?

I am attaching config files and logs for more details:
[swanctl.conf_gatewayA.txt](https://github.com/user-attachments/files/15838748/swanctl.conf_gatewayA.txt)
[swanctl.conf_gatewayB.txt](https://github.com/user-attachments/files/15838770/swanctl.conf_gatewayB.txt)
[strongswan.conf.txt](https://github.com/user-attachments/files/15838783/strongswan.conf.txt)
[docker-compose.yml_gatewayA.txt](https://github.com/user-attachments/files/15838812/docker-compose.yml_gatewayA.txt)
[docker-compose.yml_gatewayB.txt](https://github.com/user-attachments/files/15838805/docker-compose.yml_gatewayB.txt)
[gatewayA_log.txt](https://github.com/user-attachments/files/15838838/gatewayA_log.txt)

Added routes:
on gateway A:
192.168.47.0/24 dev enxac15a2985d7e proto kernel scope link src 192.168.47.198 metric 101 
and Reverse route on Gateway B.

Iptables on both gateways have FORWARD chain default policy as ACCEPT.

**Comment by tobiasbrunner** [comment]
Difficult to say. It doesn't seem like it's the strongSwan config but more the routing/firewall/whatever config of your hosts (gateway A in particular). Maybe some type of NAT?

**Reply by mkcdot** [reply]
Ok.
- Iptables on both gateways have FORWARD chain default policy as ACCEPT. No additional rules set for Iptables. (I tried adding specific FORWARD chain rules for 192.168.0.0/24 and 192.168.2.0/24 subnets. But they weren't helping. So I kept default policy as ACCEPT).
- net.ipv4.ip_forward is set to 1 in both gateways.
- Added route in gateway A:
sudo ip route add 192.168.2.0/24 via 192.168.47.199 dev enxac15a2985d7e proto static src 192.168.0.1
and reverse in gateway B.

I can provide you with routing table or iptables or any additional info for degugging. Let me know what you need to figure out this issue.

**Reply by tobiasbrunner** [reply]
strongSwan already adds such routes in table 220. And if firewalld is enabled it might configure rules via nft.

**Reply by mkcdot** [reply]
Ok. I have not found any commands for iptables or routes in this example: www.strongswan.org/testing/testresults/ikev2/net2net-cert.
So I tried to add route and iptable rules as per iptables and route output given.
Do we need to modify iptables or any network configuration for this to work? Or strongswan takes care of it?

**Reply by tobiasbrunner** [reply]
Depends completely on your needs and existing rules etc. strongSwan's [updown script](https://docs.strongswan.org/docs/5.9/plugins/updown.html) can install firewall rules, but these are only necessary if the default policies for FORWARD/INPUT are DROP/REJECT. And the script currently only support iptables, not nft. Routes are installed by strongSwan to force a source IP that's in the local traffic selector, but they might not be necessary depending on existing routes and use cases.

Oh, and also see [the docs](https://docs.strongswan.org/docs/5.9/howtos/forwarding.html) regarding forwarding traffic and possible issues with NATs.
